Zotero/Scenes/Detail/PDF/Views/PDFAnnotationsViewController.swift

```diff
@@ -164,113 +183,94 @@ final class PDFAnnotationsViewController: UIViewController {
-    private func reloadIfNeeded(for state: PDFReaderState, completion: @escaping () -> Void) {
-        if state.document.pageCount == 0 {
-            DDLogWarn("AnnotationsViewController: trying to reload empty document")
-            completion()
-            return
-        }
-
-        let isVisible = parentDelegate?.isSidebarVisible ?? false
-
-        if state.changes.contains(.annotations) {
-            if state.changes.contains(.sidebarEditing) {
-                tableView.setEditing(state.sidebarEditingEnabled, animated: false)
-            }
-            updateQueue.async { [weak self] in
-                guard let self else { return }
-                var snapshot = NSDiffableDataSourceSnapshot<Int, PDFReaderState.AnnotationKey>()
-                snapshot.appendSections([0])
-                snapshot.appendItems(state.sortedKeys)
-                if let keys = state.updatedAnnotationKeys {
-                    snapshot.reloadItems(keys)
-                }
-                dataSource.apply(snapshot, animatingDifferences: isVisible, completion: completion)
+        /// Reloads tableView if needed, based on new state. Calls completion either when reloading finished or when there was no reload.
+        /// - parameter state: Current state.
+        /// - parameter completion: Called after reload was performed or even if there was no reload.
+        func reloadIfNeeded(for state: PDFReaderState, completion: @escaping () -> Void) {
+            if state.document.pageCount == 0 {
+                DDLogWarn("AnnotationsViewController: trying to reload empty document")
+                completion()
+                return
             }
-            return
-        }
 
-        if state.changes.contains(.interfaceStyle) {
-            updateQueue.async { [weak self] in
-                guard let self else { return }
-                var snapshot = dataSource.snapshot()
-                guard !snapshot.sectionIdentifiers.isEmpty else { return }
-                snapshot.reloadSections([0])
-                dataSource.apply(snapshot, animatingDifferences: false, completion: completion)
+            let isVisible = parentDelegate?.isSidebarVisible ?? false
+
+            if state.changes.contains(.library) {
+                updateUI(state: state, forceReload: true, animatedDifferences: isVisible, completion: completion)
+                return
             }
-            return
-        }
 
-        if state.changes.contains(.selection) || state.changes.contains(.activeComment) {
-            if let keys = state.updatedAnnotationKeys {
-                updateQueue.sync { [unowned self] in
-                    var snapshot = dataSource.snapshot()
-                    snapshot.reloadItems(keys)
-                    dataSource.apply(snapshot, animatingDifferences: false)
+            if state.changes.contains(.annotations) {
+                if state.changes.contains(.sidebarEditing) {
```

Comment:
`.sidebarEditing` should be outside of this if. It can happen with any changes here, so it should always be applied.

The `reloadIfNeeded` should look like this:
```
if state.document.pageCount == 0 {
    // ...
    return
}

if state.changes.contains(.sidebarEditing) {
    tableView.setEditing(state.sidebarEditingEnabled, animated: false)
}

let isVisible = parentDelegate?.isSidebarVisible ?? false

if state.changes.contains(.library) {
    ...
    return
}

if state.changes.contains(.annotations) {
    ...
    return
}

if state.changes.contains(.interfaceStyle) && dataSource.snapshot().numberOfItems > 0 {
    ...
    return
} 

if state.changes.contains(.selection) || state.changes.contains(.activeComment) {
    ...
    return
}

completion()
```

Reply:
Check https://github.com/zotero/zotero-ios/pull/1040/commits/19acd1e7612bb3e5dcc261eeb6f3c6f0b963d740